server/src/test/java/org/elasticsearch/rest/action/RestCancellableNodeClientTests.java

```diff
@@ -303,6 +303,10 @@ public void addCloseListener(ActionListener<Void> listener) {
             // if the channel is already closed, the listener gets notified immediately, from the same thread.
             if (open.get() == false) {
                 listener.onResponse(null);
+                // Handle scenario where awaitClose() was called before any calls to addCloseListener(), this ensures closeLatch is pulled.
+                if (closeListener.isDone() == false) {
```

Comment:
Can we `assertFalse(closeListener.isDone())` on this branch too? Seems like it'd be a test bug to want to add two of them?